documentation/specs/proposed/BuildCheck.md

```diff
@@ -0,0 +1,405 @@
+* ETW for analyzers.
+* Attributing `.editorconfig` configurations to .sln files.
+* Attributing `.editorconfig` configurations to lower granularity than whole projects.
+* Respecting `.editorconfig` file in msbuild import locations (unless they are in the parent folders hieararchy of particular project file).
+ 
+
+# User Experience
+
+## Running / UX
+
+### Inbox Analyzers
+
+**TBD** - suggested list of analyzers to be shipped with V1 (along with proposed configuration)
+
+### Live Build
+
+BuildCheck will run as part of the build and execute [inbox analyzers](#inbox-analyzers) and [custom analyzers](#acquisition-of-custom-analyzers) based on the [configuration](#configuration). Users will have an option to completely opt-out from BuildCheck to run via commandline switch.
+
+Findings - reports - of analyzers will be otput as build messages/warnings/errors, while the code should help distinguish BuildCheck produced reports from regular build errors/warnings.
+
+### Binlog Replay mode
+
+Users will have option to explicitly opt-in to run BuildCheck during the binlog replay mode:
+
+```bash
+> dotnet build msbuild.binlog /analyze
+```
+
+Would there be any analyzers that are not possible to run during the replay mode (subject to internal design - this difference won't be exposed during [custom analyzers authoring](#custom-analyzers-authoring)), replay mode will inform user about those via warnings.
+
+Replay mode will by default consider `.editorconfig` files stored within the binlog and will run analyzers based on those. This may lead to double-reports – as binlog will have the runtime analysis reports stored, plus the replay-time analysis reports will be augumented.
+
+For this reason we will consider following modes (all are non-goals):
+* Ability to specify skipping all binlog stored reports
+* Ability to specify skipping of the stored .editorconfig files
+* Ability to specify single replay-time .editorconfig file and it’s precedence (only the specified, specified as most significant, specified as least significant)
+
+## Configuration
+
+There will be 3 mechanisms of configuring the analyzers and rules:
+* The default configuration declared by the analyzers themselves ([more details on implementation](#rules-declaration))
+* [Sdk Analysis Level property](https://github.com/dotnet/designs/blob/main/proposed/sdk-analysis-level.md) – mostly for the inbox analyzers
+* `.editorconfig` file
+
+For the `.editorconfig` file configuration, following will apply:
+* Only `.editorconfig` files collocated with the project file or up the folder hierarchy will be considered.
+* `.editorconfig` files placed along with explicitly or implicitly imported msbuild files won’t be considered.
+* `.editorconfig` files packaged within nuget packages won’t be considered (unless the unpack action will place them alongside the referencing project).
```

Comment:
As a rule there's no "unpack action" in modern NuGet beyond "unzip to packages folder", I'd drop that part.

Reply:
Ah OK.

Btw. there still can be custom build target doing a similar thing I suppose.. but probably not that doesn't need to be mentioned, as project folder locvation is already mentioned above..